sqlmesh/utils/metaprogramming.py

```diff
@@ -276,6 +276,14 @@ def walk(obj: t.Any) -> None:
                 build_env(v, env=env, name=k, path=path)
 
     if name not in env:
+        registry_name = getattr(obj, "registry_name", None)
+        if registry_name:
+            from sqlmesh.core.macros import macro
+
+            if registry_name == macro.registry_name:
+                # We only need to add the undecorated code of @macro() functions in env
+                obj = obj.__wrapped__
+
```

Comment:
I believe this block is going to be reached when a macro references another macro in its local scope, like how `incremental_by_ds` references `between` in the updated `macros.py`, for example. Based on this, I assumed that the `import` here shouldn't cost _too much_, but ideally I'd like to factor it out so that we don't do it over and over again (it wasn't possible to move outside due to circular `import` issues).